v2/tools/generator/internal/config/property_configuration.go

```diff
@@ -27,8 +27,9 @@ type PropertyConfiguration struct {
 	ImportConfigMapMode            configurable[ImportConfigMapMode] // The config map mode
 	IsSecret                       configurable[bool]                // Specify whether this property is a secret
 	NameInNextVersion              configurable[string]              // Name this property has in the next version
-	RenameTo                       configurable[string]              // Name this property should be renamed to
-	ResourceLifecycleOwnedByParent configurable[string]              // Name of the parent resource which owns the lifecycle of the sub-resource.
+	PayloadType                    configurable[PayloadType]
```

Comment:
Deserves a comment as well (similar to others?)

Reply:
Done.